cpp/docs/DEVELOPER_GUIDE.md

```diff
@@ -745,6 +745,22 @@ void isolated_helper_function(...);
 
 [**Anonymous namespaces should *never* be used in a header file.**](https://wiki.sei.cmu.edu/confluence/display/cplusplus/DCL59-CPP.+Do+not+define+an+unnamed+namespace+in+a+header+file)
 
+# Deprecating and Removing Code
+
+To minimize disruption for users, any breaking changes must be signaled by deprecation warnings for one release prior to the change.
+This includes wholesale removals of functions/methods/classes *and* changes to existing function signatures.
+Deprecations should be indicated using the [deprecated](https://en.cppreference.com/w/cpp/language/attributes/deprecated) attribute.
+Deprecated APIs should be documented using Doxygen's [deprecated](https://www.doxygen.nl/manual/commands.html#cmddeprecated) command.
+
+When an API is deprecated, if a replacement exists the replacement should be mentioned in both the deprecation message and the deprecation documentation.
+Changes to function parameters should be accompanied by the introduction of suitable overloads to avoid API breaks for one release cycle.
+Where possible, deprecated APIs should call new APIs, but duplication in deprecated code is acceptable at the discretion of PR reviewers if avoiding it imposes an undue burden on developers.
```

Comment:
Please adjust your word wrapping -- suggest keeping it to the same as our .cpp files -- 100 characters.

Reply:
Done.